git_pile/git_pile.py

```diff
@@ -389,6 +390,93 @@ def cmd_init(args, config):
     return 0
 
 
+class InitCmd(PileCommand):
+    """
+    Initialize configuration of an empty pile in this repository
+    """
+
+    arg("-d", "--dir", help="Directory in which to place patches (default: %(default)s)", metavar="DIR", default="patches")
```

Comment:
although I like the overall cmd split. I think this abstraction goes a little bit too far: it's an abstraction that hides too much of the implementation. I'd rather have a setup() or parse_args() methods in each *Cmd that is called by the cli. Then here we see the actual call to Argparse. In other words, I don't think we should abstract the use of Argparse per se.

Reply:
> although I like the overall cmd split. I think this abstraction goes a little bit too far: it's an abstraction that hides too much of the implementation. (...) In other words, I don't think we should abstract the use of Argparse per se.

I see `arg()` as just a convenience to keep it short as well as to save one indentation level :-). The intention was not to abstract away argparse, the documentation for the proposed module even lets the user know that `add_argument()` will be called for each call to `arg()`.

> I'd rather have a setup() or parse_args() methods in each *Cmd that is called by the cli. Then here we see the actual call to Argparse.

In this current version, we allow the user to implement the method `init()` for that purpose.

Thus:

1. Please let me know I my argument for the `arg()` shortcut has convinced you. Otherwise, I'll change it to use a initialization method as you suggested.
2. For the initialization method, we already have `init()`. Would you rather use a different name like `setup()`? (btw, I don't `parse_args()` would match what the current method is intended for).

Reply:
> * Please let me know I my argument for the `arg()` shortcut has convinced you. Otherwise, I'll change it to use a initialization method as you suggested.

IMO it's too much magic that will be hard to understand later to extend. Dunno... I think I like a simpler approach with less abstraction. But I can live with it

> * For the initialization method, we already have `init()`. Would you rather use a different name like `setup()`? (btw, I don't think `parse_args()` would match what the current method is intended for).

setup_argparse()? setup_args()?  or just leave like you did.

Reply:
> IMO it's too much magic that will be hard to understand later to extend. Dunno... I think I like a simpler approach with less abstraction.

Fair enough. I'll change it to add arguments via an initialization method.

Reply:
> setup_argparse()? setup_args()? or just leave like you did.

Okay. I'll keep using `init()` then.

Reply:
Done and force-pushed.